File: simplefactory/java/ShapeFactory.java
package io.github.androidhot.designpattern.simplefactory.java;

public class ShapeFactory {
    public static final int CIRCLE_TYPE = 0;
    public static final int SQUARE_TYPE = 1;
    public static final int TRIANGLE_TYPE = 2;

    public static Shape getShape(int type) {
        Shape subShape = null;
        switch (type) {
            case CIRCLE_TYPE:
                subShape = new Circle();
                break;
            case SQUARE_TYPE:
                subShape = new Square();
                break;
            case TRIANGLE_TYPE:
                subShape = new Triangle();
                break;
            default:
                break;
        }
        return subShape;
    }
}

File: simplefactory/kotlin/ShapeFactory.kt
package io.github.androidhot.designpattern.simplefactory.kotlin

class ShapeFactory {
    companion object {
        val CIRCLE_TYPE: Int = 0
        val SQUARE_TYPE: Int = 1
        val TRIANGLE_TYPE: Int = 2

        fun getShape(type: Int): Shape? {
            var subShape: Shape? = null
            when(type) {
                CIRCLE_TYPE -> subShape = Circle()
                SQUARE_TYPE -> subShape = Square()
                TRIANGLE_TYPE -> subShape = Triangle()
                else -> {}
            }
            return subShape
        }
    }
}
